File: src/main/java/com/lauvinson/open/assistant/configuration/SettingsConfiguration.java
package com.lauvinson.open.assistant.configuration;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.HashMap;
import java.util.Map;

/**
 * SettingsConfiguration
 * @author created by vinson on 2019/6/20
 */
@State(name = "private-interactive-assistant",storages = {@Storage(value = "$APP_CONFIG$/private-interactive-assistant_settings.xml")})
public class SettingsConfiguration implements PersistentStateComponent<SettingsConfiguration.AbilityMap> {

    @Nullable
    @Override
    public AbilityMap getState() {
        return this.abilityMap;
    }

    private AbilityMap abilityMap = new AbilityMap();

    @Override
    public void loadState(@NotNull AbilityMap abilityMap) {
        XmlSerializerUtil.copyBean(abilityMap, this.abilityMap);
    }

    public static SettingsConfiguration INSTANCE = SettingsConfiguration.getInstance();

    private static SettingsConfiguration getInstance() {
        return ServiceManager.getService(SettingsConfiguration.class);
    }

    /**
     * Ability mirror
     * @author created by vinson on 2019/6/20
     */
    static class AbilityMap {
        /** alias - api **/
        private Map<String, String> api = new HashMap<>();
    }
}

File: src/main/java/com/lauvinson/open/assistant/configuration/SettingsConfiguration.kt
package com.lauvinson.open.assistant.configuration

import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

import java.util.HashMap

/**
 * SettingsConfiguration
 * @author created by vinson on 2019/6/20
 */
@State(name = "private-interactive-assistant", storages = [Storage(value = "\$APP_CONFIG$/private-interactive-assistant_settings.xml")])
class SettingsConfiguration : PersistentStateComponent<SettingsConfiguration.AbilityMap> {

    val abilityMap = AbilityMap(LinkedHashMap())

    override fun getState(): AbilityMap? {
        return this.abilityMap
    }

    override fun loadState(abilityMap: AbilityMap) {
        XmlSerializerUtil.copyBean(abilityMap, this.abilityMap)
    }

    /**
     * Ability mirror
     * @author created by vinson on 2019/6/20
     */
    class AbilityMap(var api: LinkedHashMap<String, LinkedHashMap<String, String>>)

    companion object {

        var INSTANCE = instance

        private val instance: SettingsConfiguration
            get() = ServiceManager.getService(SettingsConfiguration::class.java)
    }
}
